File: 392-is-subsequence/392-is-subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        
      if(s.size()>t.size()) return false;
      int j=0,i=0;
      while(i<s.size() && j<t.size()){
          
          if(s[i] == t[j]){
              i++;
              j++;
          }else{
              j++;
          }
          
      }
    if(i==s.size()) return true;
    return false;
        
    }
};

File: 392-is-subsequence/392-is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        if len(s) > len(t): return False
        
        i,j=0,0
        
        while(len(s) > i and len(t) > j ):
            if(s[i] == t[j]):
                i+=1
                j+=1
            else:
                j+=1
                
        if(i == len(s)): return True
        return False
        